lightwood/api/data_source.py

```diff
@@ -331,14 +340,27 @@ def _prepare_encoders(self):
 
         for config in self.config['input_features']:
             column_name = config['name']
-            encoder_instance = self._prepare_column_encoder(
-                config,
-                is_target=False,
-                training_data=input_encoder_training_data
-            )
+            if 'previous_' in column_name:  # TODO: fragile check, should be by position
+                column_data = self.get_column_original_data(column_name)
+
+                input_encoder_training_data['previous'].append({'data': column_data,
+                                                                'name': column_name,
+                                                                'output_type': config['type']
+                                                                })
+
+        for config in self.config['input_features']:
+            column_name = config['name']
+            if 'previous_' not in column_name:  # TODO: fragile check, should be by position
```

Comment:
can just check if the column name is in `input_encoder_training_data['previous']`